beacon_node/http_metrics/src/lib.rs

```diff
@@ -39,6 +40,7 @@ pub struct Context<T: BeaconChainTypes> {
     pub chain: Option<Arc<BeaconChain<T>>>,
     pub db_path: Option<PathBuf>,
     pub freezer_db_path: Option<PathBuf>,
+    pub gossipsub_registry: Option<std::sync::Mutex<Registry>>,
```

Comment:
why `std`'s `Mutex` over `parking_lot`'s?

Reply:
I think the mutex only holds between multiple calls the to the HTTP_API. 

Wasn't sure about the mutability in reading the registry, so just used a std mutex. If we only read it in each HTTP_API call, probably better to use parking lot and take a read lock (if that is sufficent). If we are taking a write lock every time, should be the same as a usual mutex right?

Reply:
Lets revisit this and update to parking lot if we want

Reply:
Parking lot has a `Mutex` that's a drop in replacement for `std`'s, except that it's faster and errors in fewer scenarios:

https://docs.rs/parking_lot/0.11.2/parking_lot/type.Mutex.html

The `std` Mutex is also fine though, so I think we can merge as-is